plugin/pkg/admission/priority/admission.go

```diff
@@ -231,12 +231,17 @@ func (p *PriorityPlugin) getDefaultPriorityClass() (*scheduling.PriorityClass, e
 	if err != nil {
 		return nil, err
 	}
+	// In case more than one global default priority class is added as a result of a race condition,
+	// we pick the one with the lowest priority value.
+	var defaultPC *scheduling.PriorityClass
 	for _, pci := range list {
 		if pci.GlobalDefault {
-			return pci, nil
+			if defaultPC == nil || defaultPC.Value > pci.Value {
```

Comment:
Heh, this is still non-deterministic, since two names can have the same priority. Sort by lowest priority, then by name?

Reply:
But that's fine, right? We care about the value of the PriorityClass and also care whether any class exists or not. If two names have the same priority, the value that we pick will still be the same. We can also detect whether a default class exists.

Reply:
Doesn't this set the priority class on the pod? So the name it chose would be visible? Seems like we'd want the same name chosen consistently for multiple pods submitted

Reply:
Ah, looks like it doesn't set priority class, just priority

Reply:
Right, it only sets the value of priority.